WalletLibrary/WalletLibrary/RootOfTrust/Resolvers/LinkedDomainResolver.swift

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+/**
+ * Responsible for resolving the Root of Trust through the Linked Domain.
+ */
+struct LinkedDomainResolver: RootOfTrustResolver
+{
+    private let linkedDomainService: LinkedDomainService
+    
+    private let configuration: LibraryConfiguration
+    
+    init(configuration: LibraryConfiguration)
+    {
+        self.configuration = configuration
+        
+        if let networking = configuration.networking as? WalletLibraryNetworking
+        {
+            self.linkedDomainService = LinkedDomainService(correlationVector: networking.correlationHeader,
+                                                           urlSession: networking.urlSession)
+        }
+        else
+        {
+            self.linkedDomainService = LinkedDomainService(urlSession: URLSession.shared)
+        }
+    }
+    
+    /// Resolves the Root of Trust through Linked Domains.
+    /// - Parameters:
+    ///   - identifier: An identifier used to resolve the Linked Domains. This should be the `IdentifierDocument` in this case.
+    func resolve(using identifier: Any) async throws -> RootOfTrust
+    {
+        guard let identifier = identifier as? IdentifierDocument else
+        {
+            throw VerifiedIdError(message: "", code: "")
```

Comment:
Not a super useful error message. Why is the identifier an `Any` here? Internal SDK class that we can't reference without making public?

Reply:
oh that was an accident. let me fix.